### PR TITLE
Add allowAnyValue

### PR DESCRIPTION
**Pull request by stumper66**
Added a new option: allowAnyValue
This allows me to use ListArgumentBuilder for complex tab suggestions and simply passing all input so I can do my own validation.

This allowed me to create this command: https://youtu.be/q03_paw9Vzk

### CODE REVIEW

File: docssrc/src/argument_list.md [hunk 1]
```diff
@@ -57,6 +57,20 @@ $$\downarrow$$
 
 $$\downarrow$$
 
+> ### Allowing any value (Optional)
+>
+> If you want to do your own validation you can enable the `allowAnyValue` method to set whether any value is allowed. By default, this is disabled.
+>
+> When `allowAnyValue` is enabled, validation is skipped, and it is expected you will manually perform any validation required:
```
Comment (DerEchtePilz) [inline docssrc/src/argument_list.md:64]:
```suggestion
> When `allowAnyValue` is set to `true`, list validation is skipped, and any value a player enters will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too:
```
Maybe this is a bit better?

File: docssrc/src/argument_list.md [hunk 2]
```diff
@@ -57,6 +57,20 @@ $$\downarrow$$
...
+>
+> ```java
+> ListArgumentBuilder.allowAnyValue(true);
+> ```
+>
+> ![List arguments with anyvalue enabled](./images/arguments/listargument_allow_any_values.gif)
```
Comment (DerEchtePilz) [inline docssrc/src/argument_list.md:70]:
```suggestion
> ![List arguments with allowAnyValue set to true](./images/arguments/listargument_allow_any_values.gif)
```

**Review by DerEchtePilz — CHANGES_REQUESTED** [review]
Just some suggestions I have but nothing that necessarily has to happen if we think the current state is good enough.

File: docssrc/src/argument_list.md (hunk 1 above)
Comment (JorelAli) [reply docssrc/src/argument_list.md:64]: I don't particularly like the phrase "list validation", we could just be clear as day and remove it. By saying "... is set to `true`, any value a player enters ...", it's so much cleaer.

File: docssrc/src/argument_list.md [hunk 3]
```diff
@@ -57,6 +57,20 @@ $$\downarrow$$
 
 $$\downarrow$$
 
+> ### Allowing any value (Optional)
+>
+> If you want to disable list validation you can enable the `allowAnyValue` method to set whether any value is allowed. By default, this is disabled.
+>
+> When `allowAnyValue` is set to `true`, any value a player enter will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too
```
Comment (DerEchtePilz) [inline docssrc/src/argument_list.md:64]:
```suggestion
> When `allowAnyValue` is set to `true`, any value a player enters will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too
```

Comment (JorelAli) [reply docssrc/src/argument_list.md:64]: Full stop?

Comment (DerEchtePilz) [reply docssrc/src/argument_list.md]: No, needs a colon here, I forgot that

**Review by DerEchtePilz — APPROVED** [review]
Approving this for now, but I still want to run some more manual tests before merging to make sure it all works correctly.

**Comment by DerEchtePilz** [comment]
Hmm, continuing with the test command I had from #570, when I execute this command
```java
/listtest skip a c e f b d w r t
```

I get the `Duplicate values are not allowed` error message so something still seems to be missing.

**Pull request closed without merging**